cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -174,20 +174,6 @@ func upgradeComponent(component string, waiter apiclient.Waiter, pathMgr StaticP
 		recoverEtcd = true
 	}
 
-	// We currently depend on getting the Etcd mirror Pod hash from the KubeAPIServer;
-	// Upgrading the Etcd protocol takes down the apiserver, so we can't verify component restarts if we restart Etcd independently.
-	// Skip waiting for Etcd to restart and immediately move on to updating the apiserver.
-	if component == constants.Etcd {
-		waitForComponentRestart = false
```

Comment:
If this piece of code goes away I think that the `waitForComponentRestart` var could go away too and the [if](https://github.com/kubernetes/kubernetes/blob/2ea50003797952e599ae24cd58e6e9f98caa4d5a/cmd/kubeadm/app/phases/upgrade/staticpods.go#L228) statement below in the same func could be simplified throwing away the else branch